libs/libcommon/include/common/Pipe.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdexcept>
+
+struct LazyPipe
+{
+    int fds_rw[2] = {-1, -1};
+
+    LazyPipe() = default;
+
+    virtual void open()
```

Comment:
Function is virtual but I don't see where it is overrided.
If it isn't - don't make it virtual.